lib/ansible/galaxy/api.py

```diff
@@ -46,6 +53,15 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def rate_limit_exception(exception=None):
```

Comment:
Any particular reason to have a default? Also, this looks like a test-function / predicate. So it's better to prefix it with `is_` to communicate the action it performs.

Reply:
For a while (while trying to refactor `reboot`) this function was actually called as the `except retry_condition():` and the exception was retrieved from `sys`.

Reply:
I think adopting the interface outlined @ https://github.com/ansible/ansible/pull/74240/files#r615180154 should accommodate this case.

Reply:
It looks like it could accommodate it in the same way as this implementation could - which is, not well. `giveup` (or in this implementation, `retry_condition`) would need to accept the delay in addition to the exception (and args/kwargs additional to those used by the function being retried) and it would do the extra stuff as a side-effect when `giveup` would return False (or in this implementation, when retry_condition return True). I'm still poking at `reboot` (maybe I should add a `do_on_retry` callable?), but it seems like an excessive special case.

Reply:
I added a do_on_retry optional parameter. Here's what the reboot module could look like refactored to use the common code: https://gist.github.com/s-hertel/b87bad21adea00e4a91fd68a307e5fc6

Reply:
That looks nice (although I have comments on the complexity/nesting but that's unrelated)

Reply:
:smile: I'll keep that in mind and try to simplify when I open the PR.